### PR TITLE
v0.1.0 - A replacement for chef_client_scheduler

### PR DESCRIPTION
**Pull request by RoboticCheese**
With the initial goal of replicating as much of the current client config as possible. To do:

- [x] Take over management of cron config (or decide to switch to running as a daemon)
- [x] " logrotate config
- [x] " Chef helper scripts

### CODE REVIEW

**Comment by RoboticCheese** [comment]
A `client.rb` built with `chef_client_scheduler`:

```
enable_reporting            false
log_level                   :auto
log_location                "/var/log/chef-client.log"
node_name                   "i-8335c55a.bastion.us-west-2a.aws-us-west-2-staging.socrata.net"
chef_server_url             "https://chef.socrata.net/organizations/socrata"
validation_client_name      "socrata-validator"
validation_key              "/etc/chef/validation.pem"
environment                 "aws-us-west-2-staging"
no_lazy_load                true                        # we should remove this on upgrade to chef 12
```

A `client.rb` built with `socrata-chef-client`:

```
chef_server_url "http://127.0.0.1:4545"
client_fork true
enable_reporting false
environment "test_environment"
log_level :auto
log_location "/var/log/chef/client.log"
no_lazy_load true
validation_client_name "chef-validator"
verify_api_cert true
node_name "vagrant"



# Do no crash if a handler is missing / not installed yet
begin
rescue NameError => e
  Chef::Log.error e
end

Dir.glob(File.join("/etc/chef", "client.d", "*.rb")).each do |conf|
  Chef::Config.from_file(conf)
end
```

**Comment by RoboticCheese** [comment]
A crontab entry built by `chef_client_scheduler`:

```
0,20,40 * * * * [ ! -f /var/lock/chef-client ] && /usr/bin/chef-client -L /var/log/chef/client.log -s 900 >/dev/null 2>&1
```

A cron job file built by `socrata-chef-client`:

```
0,20,40 * * * * root /bin/sleep 736;  /usr/bin/chef-client >> /var/log/chef/client.log 2>&1
```

File: attributes/default.rb [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+  # c['node_name']
+  # c['chef_server_url']
+  # c['validation_client_name']
+
+  # The validation key is only needed for the initial bootstrap.
+  # c['validation_key']
+
+  c['environment'] = Chef::Config[:environment]
+
+  c['no_lazy_load'] = true if Chef::VERSION.to_i < 12
```
Comment (jhaynes) [inline attributes/default.rb]: Am I understanding ruby correctly? In the case that the condition fails, this attribute will be nil (unless it has been set by something else?).
```irb(main):001:0> x = true if true
=> true
irb(main):002:0> x = true if false
=> nil
irb(main):003:0> x
=> true
irb(main):004:0> y = true if false
=> nil
irb(main):005:0> y
=> nil
irb(main):006:0>```

Comment (RoboticCheese) [inline attributes/default.rb]: Wow. That is super weird.

```
irb(main):001:0> x
NameError: undefined local variable or method `x' for main:Object
	from (irb):1
	from /usr/local/opt/ruby/bin/irb:11:in `<main>'
irb(main):002:0> if false
irb(main):003:1>   x = true
irb(main):004:1> end
=> nil
irb(main):005:0> x
=> nil
irb(main):006:0>
```

Looks like that's only for the variables themselves, though, not hash keys:

```
irb(main):001:0> x
NameError: undefined local variable or method `x' for main:Object
	from (irb):1
	from /usr/local/opt/ruby/bin/irb:11:in `<main>'
irb(main):002:0> x = {}
=> {}
irb(main):003:0> x['pants'] = 'jeans' if false
=> nil
irb(main):004:0> x
=> {}
irb(main):005:0> x.keys
=> []
```

Comment (RoboticCheese) [inline attributes/default.rb]: Did a Chef 12 Kitchen build just now to verify, and it correctly renders a client.rb without a `no_lazy_load` line.

Comment (jhaynes) [inline attributes/default.rb]: :shipit: ... but yeah, ruby is weird.

**Pull request merged**